packit/config.py

```diff
@@ -151,6 +152,55 @@ def is_dict_valid(cls, raw_dict: dict) -> bool:
         return Draft4Validator(JOB_CONFIG_SCHEMA).is_valid(raw_dict)
 
 
+class SyncFilesItem(NamedTuple):
+    src: str
+    dest: str
+
+
+class SyncFilesConfig:
+    def __init__(self, files_to_sync: List[SyncFilesItem]):
+        self.files_to_sync = files_to_sync
+
+    @classmethod
+    def get_from_dict(cls, raw_dict: dict, validate=True) -> "SyncFilesConfig":
+        if validate and not SyncFilesConfig.is_dict_valid(raw_dict):
+            raise Exception(f"Sync files config not valid.")
+
+        files_to_sync = []
+        if isinstance(raw_dict, list):
+            for f in raw_dict:
+                if isinstance(f, dict):
+                    files_to_sync.append(SyncFilesItem(src=f["src"], dest=f["dest"]))
+                else:
+                    files_to_sync.append(SyncFilesItem(src=f, dest=f))
+        if isinstance(raw_dict, dict):
+            for f in raw_dict:
+                files_to_sync.append(SyncFilesItem(src=f["src"], dest=f["dest"]))
+
+        return SyncFilesConfig(files_to_sync=files_to_sync)
+
+    @classmethod
+    def is_dict_valid(cls, raw_dict: dict) -> bool:
+        return Draft4Validator(SYNCED_FILES_SCHEMA).is_valid(raw_dict)
+
+    def __eq__(self, other: object):
+        if not isinstance(other, SyncFilesConfig):
+            return NotImplemented
+
+        if not self.files_to_sync and not other.files_to_sync:
+            return True
+
+        if len(self.files_to_sync) != len(other.files_to_sync):
+            return False
+        compare = False
+        for f in self.files_to_sync:
+            for o in other.files_to_sync:
+                if f.src == o.src and f.dest == o.dest:
+                    compare = True
```

Comment:
Hmm, this seems to return `True` if ANY file from one object is equal to ANY file from the other object.
I'd expect that it returns `True` if ALL files are equal.

Reply:
If you implement `__eq__()` in `SyncFilesItem` you should be able to just:

```python
return self.files_to_sync == other.files_to_sync
```